src/renderer/qssWidget/js/qssBaseStepperWidget.js

```diff
@@ -0,0 +1,528 @@
+        var animationClasses = fluid.values(styles);
+        animatedElements.removeClass(animationClasses.join(" "));
+    };
+
+    /**
+     * Applies an animation to an increase/decrease button. The type of animation depends
+     * on whether the value was increased properly or a lower/upper bound has been reached.
+     * @param {Object} styles - An object whose values are the CSS animation classes.
+     * @param {jQuery} button - A jQuery object representing the items to be animated.
+     * @param {Boolean} isError - Whether the upper/lower bound has been reached or not.
+     */
+    gpii.qssWidget.baseStepper.animateButton = function (styles, button, isError) {
+        var triggerClass = isError ? styles.errorAnimation : styles.warningAnimation;
+
+        gpii.qssWidget.baseStepper.triggerCssAnimation(
+            button,
+            triggerClass,
+            [ styles.errorAnimation, styles.warningAnimation ]);
+    };
+
+    /**
+     * Creates and manages the setting "indicators" list.
+     */
+    fluid.defaults("gpii.qssWidget.baseStepper.indicators", {
+        gradeNames: "gpii.psp.repeater",
+
+        //
+        // Repeater stuff
+        //
+        dynamicContainerMarkup: {
+            container: "<div role='radio' class='%containerClass fl-qssStepperWidget-indicator' tabindex='-1'></div>",
+            containerClassPrefix: "flc-qssStepperWidget-indicator"
+        },
+        handlerType: "gpii.qssWidget.baseStepper.indicator.presenter",
+        markup: null,
+
+        //
+        // Custom
+        //
+        model: {
+            setting: {}
+        },
+        // Build the repeater items
+        modelRelay: {
+            items: {
+                target: "items",
+                singleTransform: {
+                    type: "fluid.transforms.free",
+                    func: "gpii.qssWidget.baseStepper.getIndicatorsList",
+                    args: [
+                        "{gpii.qssWidget.baseStepper.indicators}.model.setting"
+                    ]
+                }
+            }
+        },
+
+        events: {
+            onIndicatorClicked: null
+        },
+
+        listeners: {
+            "onIndicatorClicked.updateValue": {
+                changePath: "setting.value",
+                value: "{arguments}.0",
+                source: "fromWidget"
+            }
+        }
+    });
+
+
+    /**
+     * Generates the different indicators' data based on a setting .
+     * indicators are generated using the setting's `min`, `max` and `divisibleBy` properties.
+     * In case either of those is missing, no indicators will be generated.
+     * Note that items will be recomputed every time the setting changes but only items that
+     * need to be re-rendered will do so (changeApplier merges the values).
+     * @param {Object} setting - The setting for which indicators must be created
+     * @return {Object[]} - The list of data for each indicator element. In case no indicators
```

Comment:
Please supply an @typedef for this structure so it is clear what data actually gets returned